Add syft updater

NikolaMilosavljevic opened the pull request:
Fixes: https://github.com/dotnet/dotnet-docker/issues/4545

Adds a simple syft updater.

lbussell reviewed (changes requested):
Please base this off of nightly - the update dependencies code for updating based off of GitHub releases has changed substantially in the past month.

It has not made it into main yet since servicing is next week and the tool is never called from main.

NikolaMilosavljevic commented:
> Please base this off of nightly - the update dependencies code for updating based off of GitHub releases has changed substantially in the past month.
> 
> It has not made it into main yet since servicing is next week and the tool is never called from main.

Fixed with https://github.com/dotnet/dotnet-docker/pull/5769/commits/ad5d4e5cd5f8e6d984ec2ac38e640986a58cb27f

I've rebased and force-pushed the change.

lbussell commented on eng/update-dependencies/SyftUpdater.cs:
```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text.RegularExpressions;
```
```suggestion
```

These imports appear to be unused.

NikolaMilosavljevic replied:
Fixed with https://github.com/dotnet/dotnet-docker/pull/5769/commits/810c6cd1a6e65ea350b4878c464d2d831217e9ca
